tools/layoutflash/src/main.rs

```diff
@@ -67,9 +67,16 @@ fn read_fixed_fdt(path: &Path) -> io::Result<Vec<Area>> {
     Ok(areas)
 }
 
+// This method assumes that areas are sorted by offset.
 fn layout_flash(path: &Path, areas: &[Area]) -> io::Result<()> {
     let mut f = fs::File::create(path)?;
+    let mut last_area_end = 0;
     for a in areas {
+        if a.offset < last_area_end {
```

Comment:
Why do they need to be sorted by offset? Why place that burden on the user? I can tell you it gets old fast. 

Option 1: read in the file, sort it, then do this check.
Option 2: provide a tool that reads the file in and spits it out sorted (I kind of like that one)

Please don't make people hand sort files. It's a pain.

Reply:
I am sorry about it, added sorting. I was thinking about adding sorting, and I did not do it initially because:

* it is slower (I know it is silly reason, especially that areas has only few elements)
* I was kind of preparing for trying to tackle last point in #167, where I believe the intent is to drop the offset and just auto-compute it based on size, and for that areas have to be sorted in file. But I should not have made this decision based on some future plan (which I have a little mixed feelings about and I am not sure if it will even be approved). 

I think it is a right decision to do the sorting here now and we can always drop it in the future if it will not be needed.